api/plugin/file/fileutils.go

```diff
@@ -0,0 +1,65 @@
+package file
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func getRealPath(path string) string {
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal("Error getting working directory:", err)
+	}
+	return filepath.Join(wd[:len(wd)-23], path)
+}
+
+// A valid path should be a shortest path name equivalent to path by purely lexical processingand.
+// Specifically, it should start with "/parigot/app/", also, any use of '.', '..', in the path is
+// not allowed.
+func isValidPath(fpath string) (string, bool) {
+	fileName := filepath.Base(fpath)
+	dir := strings.ReplaceAll(fpath, fileName, "")
+	if !strings.HasPrefix(dir, pathPrefix) || strings.Contains(dir, ".") {
+		return fpath, false
+	}
+	cleanPath := filepath.Clean(fpath)
+
+	return cleanPath, true
+}
+
+func isValidBuf(buf []byte) bool { return len(buf) <= maxBufSize }
+
+// Deletes the specified file and its parent directories if they're empty
+func deleteFileAndParentDirIfNeeded(path string) {
+	realPath := getRealPath(path)
+
+	// Delete the file
+	err := os.Remove(realPath)
```

Comment:
Do you have a definition for what the file service SHOULD do if you try to delete a file that is currently open? Once you decide what you want to do there, you should write a test to insure the behavior you want is enforced.

Reply:
It will check the status of the file before actually deleting it, so only closed files will be deleted. I'm working on this now